akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -239,6 +240,21 @@ object HttpEntity {
       else Default(contentType, data.length, limitableByteSource(Source.single(data))) withSizeLimit maxBytes
 
     override def productPrefix = "HttpEntity.Strict"
+
+    override def toString = {
+      val dataAsString = contentType match {
+        case _: Binary ⇒
+          data.toString()
+        case nb: NonBinary ⇒
+          val decodedString = data.decodeString(nb.charset.value)
+          if (decodedString.length > 4000)
```

Comment:
We would, then, become unable to report the number of bytes remaining (since we didn't know how many bytes the last character contained). Is that fine nevertheless?